File: ETankapplicationRESTAPI/ETankRESTAPI/src/main/java/thl/gruppef/etankrest/etankrestapi/entities/UserSettings.java
package thl.gruppef.etankrest.etankrestapi.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

/**
 * Describes the database entity for user settings
 */
@Entity
@Getter
@Setter
public class UserSettings {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    //Ohne Ignore läuft er in eine endlosschleife
    @JsonIgnore
    @OneToOne(mappedBy = "userSettings")
    private User userRef;

    private int gameSoundVolume;

    private int gameMusicVolume;

    private boolean gameSoundOn;

    private boolean gameMusicOn;

    private String showStatisticKey;

    private String moveUpKey;

    private String moveLeftKey;

    private String moveRightKey;

    private String moveDownKey;

    private String fireMainWeaponKey;

    private String fireSecondaryWeaponKey;

    /**
     * Constructor of the class
     * with default values
     */
    public UserSettings() {
        this.gameSoundVolume = 100;
        this.gameMusicVolume = 100;
        this.gameSoundOn = true;
        this.gameMusicOn = true;
        this.showStatisticKey = "TAB";
        this.moveUpKey = "W";
        this.moveLeftKey = "A";
        this.moveRightKey = "D";
        this.moveDownKey = "S";
        this.fireMainWeaponKey = "SPACE";
        this.fireSecondaryWeaponKey = "COMMAND";
    }
}

File: client_python/src/model/data/UserSettings.py
import json


class UserSettings:
    def __init__(self):
        self.id = 1
        self.gameSoundVolume = 100
        self.gameMusicVolume = 100
        self.gameSoundOn = True
        self.gameMusicOn = True
        self.moveUpKey = "W"
        self.moveLeftKey = "A"
        self.moveDownKey = "S"
        self.moveRightKey = "D"
        self.fireMainWeaponKey = "SPACE"
        self.fireSecondaryWeaponKey = "CTL"
        self.showStatisticKey = "TAB"

    def buildSettingJson(self):
        settingsDict = {
            "gameSoundVolume": self.gameSoundVolume,
            "gameMusicVolume": self.gameMusicVolume,
            "gameSoundOn": self.gameSoundOn,
            "gameMusicOn": self.gameMusicOn,
            "moveUpKey": self.moveUpKey,
            "moveLeftKey": self.moveLeftKey,
            "moveDownKey": self.moveDownKey,
            "moveRightKey": self.moveRightKey,
            "fireMainWeaponKey": self.fireMainWeaponKey,
            "fireSecondaryWeaponKey": self.fireSecondaryWeaponKey,
            "showStatisticKey": self.showStatisticKey
        }
        return json.dumps(settingsDict)
